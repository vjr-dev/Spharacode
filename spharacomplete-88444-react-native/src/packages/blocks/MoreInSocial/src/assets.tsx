// Customizable Area Start
export const back1 = require("../assets/image_background-1.png");
export const back2 = require("../assets/image_background-2.png");
export const call = require("../assets/call-answer.png");
export const chatboat = require("../assets/chatbot.png");
export const files = require("../assets/files.png");
export const alerts = require("../assets/my_alerts.png");
export const notifications = require("../assets/notifications.png");
export const saved = require("../assets/saved.png");
export const settings = require("../assets/settings.png");
// Customizable Area End