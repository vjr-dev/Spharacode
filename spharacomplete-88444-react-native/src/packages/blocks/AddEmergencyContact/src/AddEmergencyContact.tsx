// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";
import {
  Image,
  ImageBackground,
  SafeAreaView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  ScrollView,
  Modal,
  ActivityIndicator
} from "react-native";
import { COLORS } from "framework/src/Globals";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";
import AddEmergencyContactController, {
  Props,
} from "./AddEmergencyContactController";
import { back1, back2, image_back, Profile } from "./assets";
import { scaledSize } from "../../../framework/src/Utilities";
import { deviceHeight, deviceWidth } from "framework/src/Utilities";
import { displayConfirmAlert } from "../../../components/src/CustomAlert";

export default class AddEmergencyContact extends AddEmergencyContactController {
  constructor(props: Props) {
    super(props);
  }

  getContactType = (type) => {
    if( type === 'emergency_contact'){
      return "Emergency Contact"
    }else if(type === 'friends'){
      return 'Friends'
    }else{
      return 'Family'
    }
  }
  renderItem = (item) => {
    return (
      <View style={styles.itemView}>
        <View style={ styles.itemInnerView}>
          <Image
            source={Profile}
            style={styles.profileImage}
          />
          <View style={styles.textView}>
            <View style={{flex:1}}>
              <Text style={styles.name} numberOfLines={1}>{item.attributes.name}</Text>
              <Text testID={"phoneNumberText"} style={styles.number} numberOfLines={1}>{item.attributes.phone_number}</Text>
              <Text style={styles.contactType} numberOfLines={1}>{this.getContactType(item.attributes.contact_type)} </Text>
            </View>
            {this.state.ListData.length == 1 ? null : (
            <TouchableOpacity
              testID="addEmergencyContactDeleteButton"
              onPress={() => this.deleteItemConfirmation(item)}
              style={styles.deleteButton}
            >
              <MaterialIcon name="close" color={COLORS.black} size={scaledSize(17)} />
            </TouchableOpacity>
          )}
          </View>
        </View>
      </View>
    );
  };
  deleteItemConfirmation(item: any) {
    displayConfirmAlert("Are you sure?","Do you want to Delete Emergency contact?", ()=> this.deleteContacts(item.id))
  }
  render() {
    return (
      <SafeAreaView style={styles.container}>
        <ImageBackground source={back1} style={styles.bgImage}>
          <ImageBackground source={back2} style={styles.bgImage}>
            <View style= {styles.body}>
              <View style={styles.headerView} >
                <TouchableOpacity
                  testID="addEmergencyContactBackButton"
                  style={styles.backButton}
                  onPress={() => this.handleBackButtonClick()}
                >
                  <Image
                    source={image_back}
                    style={styles.backIcon}
                  />
                </TouchableOpacity>
                
                <TouchableOpacity
                  testID="addEmergencyContactOpenAddNewModalButton"
                  onPress={() => this.onAddNewContacts()}
                  style={styles.addNewButton}
                >
                  <Text style={styles.addNewLabel}> ADD NEW </Text>
                </TouchableOpacity>
              </View>
              <Text style={styles.title}> Add Emergency Contacts.</Text>
              <Text style={styles.label}>
                Your emergency contacts will receive your message  when you trigger the panic button.
              </Text>
              <View style={styles.contactView}>
                <ScrollView showsVerticalScrollIndicator={false}>
                  {
                    this.state.ListData.map((item: any)=>{
                      return (this.renderItem(item));
                    })
                  }
                </ScrollView>
              </View>
              <TouchableOpacity
                testID="addEmergencyContactDoneButton"
                onPress={() => {
                  this.onDoneClick();
                }}
                style={styles.doneButton}
              >
                <Text style={{ color: "#fff", fontWeight: "bold" }}>DONE</Text>
              </TouchableOpacity>
            </View>
            <Modal visible={this.state.isLoading} transparent={true}>
              <View style={styles.loaderContainer}>
                <ActivityIndicator
                  animating={true}
                  color="#f07233"
                  hidesWhenStopped={true}
                  size="large"
                />
              </View>
            </Modal>
          </ImageBackground>
        </ImageBackground>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1, 
    backgroundColor: "#454545"
  },
  bgImage: {
    height: deviceHeight,
    width: deviceWidth,
    flex: 1,
  },
  body:{
    flex:1,
    padding: scaledSize(10),
  },
  headerView:{
    height: scaledSize(30),
    width: "100%",
    marginVertical: scaledSize(10),
    justifyContent: "space-between",
    alignItems:'center',
    flexDirection: "row",
  },
  backButton:{
    height: scaledSize(30), 
    width: scaledSize(30), 
    justifyContent:'center',
    alignItems:'center'
  },
  backIcon:{
    height: scaledSize(15), 
    width: scaledSize(15), 
    tintColor: "#f07233"
  },
  addNewButton:{
    height: scaledSize(30), 
    paddingHorizontal: scaledSize(10),
    justifyContent:'center',
    alignItems:'center'
  },
  addNewLabel:{
    color: "#f07233", 
    fontSize: scaledSize(13), 
  },
  title:{
    fontSize: scaledSize(20),
    color: "#fff",
  },
  label:{
    fontSize: scaledSize(12),
    color: "#cdcdcd",
    marginTop: scaledSize(10),
  },
  contactView:{
    flex:1,
    paddingVertical: scaledSize(10)
  },
  doneButton:{
    height: scaledSize(40),
    width: "100%",
    marginBottom: scaledSize(15),
    backgroundColor: "#f07233",
    borderRadius: scaledSize(20),
    alignItems: "center",
    justifyContent: "center",
  },
  itemView:{
    alignSelf: "center",
    marginVertical: scaledSize(5),
    alignItems: "center",
  },
  itemInnerView:{
    flexDirection: "row",
    alignItems:'center',
    justifyContent:'center'
  },
  profileImage:{
    height: scaledSize(40),
    width: scaledSize(40),
    marginRight: scaledSize(10),
    borderRadius: scaledSize(20),
  },
  textView:{
    flex:1,
    flexDirection:'row',
    alignItems:'center',
    justifyContent:'space-between',
    paddingBottom: scaledSize(5),
    borderBottomWidth: 0.5,
    borderColor:"#adadad"
  },
  name:{
    color: "#adadad", 
    fontSize: scaledSize(12)
  },
  number:{
    color: "#adadad", 
    fontSize: scaledSize(11)
  },
  contactType:{
    color: "#adadad", 
    fontSize: scaledSize(10)
  },
  deleteButton:{
    height: scaledSize(20),
    width: scaledSize(20),
    marginRight: scaledSize(20),
    marginLeft: scaledSize(10),
    backgroundColor: "#adadad",
    alignItems: "center",
    justifyContent: "center",
    borderRadius: scaledSize(10),
  },
  loaderContainer:{
    flex:1,
    alignItems:'center',
    justifyContent:'center'
  }
});
// Customizable Area End
