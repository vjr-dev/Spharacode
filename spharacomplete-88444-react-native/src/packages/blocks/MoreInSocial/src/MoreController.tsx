// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import { BlockComponent } from "framework/src/BlockComponent";
import { IBlock } from "framework/src/IBlock";
import { Message } from "framework/src/Message";
import { runEngine } from "framework/src/RunEngine";

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
}

interface S {
  modalVisible: boolean;
}

interface SS {
  id: any;
}

export default class MoreController extends BlockComponent<Props, S, SS> {
  focusListener: any;

  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);
    this.subScribedMessages = [];

    this.state = {
      modalVisible: false,
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
  async componentDidMount() {
    this.focusListener = this.props.navigation.addListener("focus", () => {
      this.setState({
        modalVisible: false,
      });
    });
  }
}
// Customizable Area End
