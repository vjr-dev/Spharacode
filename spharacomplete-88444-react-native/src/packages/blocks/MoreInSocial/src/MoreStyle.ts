// Customizable Area Start

import { Dimensions, StyleSheet } from "react-native";
import { COLORS } from "framework/src/Globals";
const sh = Dimensions.get("window").height;
const sw = Dimensions.get("window").width;

export const Styles = StyleSheet.create({
  container: {
    height: sh,
    width: sw,
    flex: 1,
  },
  safeareaView: {
    flex: 1,
    backgroundColor: "#454545",
  },
  modalView: {
    position: "absolute",
    bottom: 0,
    width: sw,
    alignSelf: "center",
    backgroundColor: "#4a4a4a",
  },
  shadowView: {
    backgroundColor: "#3e3e3e",
    width: sw,
    height: sh,
    position: "absolute",
    opacity: 0.6,
    flex: 1,
  },
  menuContainer: {
    flexWrap: "wrap",
    marginBottom: 23,
    alignItems: "flex-start",
    flexDirection: "row",
  },
  menuButton: {
    marginTop: 23,
    justifyContent: "center",
    alignItems: "center",
  },
  menuImage: {
    width: 20,
    height: 20,
  },
  menuText: {
    fontSize: 13,
    paddingTop: 5,
    paddingHorizontal: 23,
    color: COLORS.infoGray,
  },
});
// Customizable Area End
