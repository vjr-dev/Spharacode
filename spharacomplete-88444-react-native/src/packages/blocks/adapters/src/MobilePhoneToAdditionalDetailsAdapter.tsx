import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
