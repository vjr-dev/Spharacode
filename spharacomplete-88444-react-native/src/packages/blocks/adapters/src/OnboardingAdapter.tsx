import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class OnboardingAdapter {
  send: (message: Message) => void;

  constructor() {
    // const uuidv4 = require('uuid/v4');
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.OnboardingGuideControllerDoneOnboarding),
      getName(MessageEnum.OnboardingGuideControllerSkipOnboarding)
    ]);
  }

  convert = (from: Message): Message => {
    runEngine.debugLog("OnboardingAdapter", from.id)
    const to = new Message(getName(MessageEnum.NavigationMessage));
    to.addData(getName(MessageEnum.NavigationTargetMessage), 'Home');
    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );
    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
