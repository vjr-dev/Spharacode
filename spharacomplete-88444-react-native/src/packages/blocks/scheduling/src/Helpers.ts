/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});