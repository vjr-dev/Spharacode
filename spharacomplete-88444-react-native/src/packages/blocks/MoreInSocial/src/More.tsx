// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";
import { Component } from "react";
import {
  Image,
  ImageBackground,
  ImageSourcePropType,
  SafeAreaView,
  Text,
  TouchableOpacity,
  View,
} from "react-native";

import * as IMAGE from "./assets";
import MoreController, { Props } from "./MoreController";
import { Styles } from "./MoreStyle";

export default class More extends MoreController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <SafeAreaView style={Styles.safeareaView}>
        <ImageBackground source={IMAGE.back1} style={Styles.container}>
          <ImageBackground source={IMAGE.back2} style={Styles.container}>
            <Text style={{ color: "white", paddingTop: 20 }}>More Tab</Text>
            <TouchableOpacity
              testID="moreInSocialModalButton"
              style={{ marginTop: 20 }}
              onPress={() => this.setState({ modalVisible: true })}
            >
              <Text style={{ color: "white" }}>Open Modal</Text>
            </TouchableOpacity>
            {this.state.modalVisible ? (
              <View style={Styles.shadowView} />
            ) : null}
            {this.state.modalVisible ? (
              <View style={{ flex: 1 }}>
                <View style={Styles.modalView}>
                  <View style={Styles.menuContainer}>
                    <MenuComponent
                      testID="callMenu"
                      onMenuPress={() => {}}
                      path={IMAGE.call}
                      text="Calls"
                    />
                    <MenuComponent
                      testID="botMenu"
                      onMenuPress={() => {}}
                      path={IMAGE.chatboat}
                      text="Bot"
                    />
                    <MenuComponent
                      testID="fileMenu"
                      onMenuPress={() => {}}
                      path={IMAGE.files}
                      text="Files"
                    />
                    <MenuComponent
                      testID="savedMenu"
                      onMenuPress={() => {}}
                      path={IMAGE.saved}
                      text="Saved"
                    />
                    <MenuComponent
                      testID="alertsMenu"
                      onMenuPress={() => {}}
                      path={IMAGE.alerts}
                      text="My Alerts"
                    />
                    <MenuComponent
                      testID="notificationMenu"
                      onMenuPress={() => {}}
                      path={IMAGE.notifications}
                      text="Notifications"
                    />
                    <MenuComponent
                      testID="settingslMenu"
                      onMenuPress={() => {}}
                      path={IMAGE.settings}
                      text="Settings"
                    />
                  </View>
                </View>
              </View>
            ) : null}
          </ImageBackground>
        </ImageBackground>
      </SafeAreaView>
    );
  }
}
export interface MenuPropTypes {
  path: ImageSourcePropType;
  text: string;
  onMenuPress: () => void;
}
export class MenuComponent extends Component<MenuPropTypes> {
  constructor(props: MenuPropTypes) {
    super(props);
  }

  render() {
    return (
      <TouchableOpacity
        testID="onMenuPreeButton"
        onPress={this.props.onMenuPress}
        style={Styles.menuButton}
      >
        <Image
          source={this.props.path}
          resizeMode="contain"
          style={Styles.menuImage}
        />
        <Text style={Styles.menuText}>{this.props.text}</Text>
      </TouchableOpacity>
    );
  }
}
// Customizable Area End
