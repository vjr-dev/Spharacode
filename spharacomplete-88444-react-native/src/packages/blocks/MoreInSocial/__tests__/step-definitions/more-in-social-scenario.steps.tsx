//@ts-nocheck
//@ts-ignore
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import React from "react";
import More from "../../src/More";
import { MenuComponent } from "../../src/More";

const screenProps = {
  navigation: {
    dispatch: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  id: "MoreInSocial",
};

const feature = loadFeature(
  "./__tests__/features/more-in-social-scenario.feature"
);

defineFeature(feature, (test) => {
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({ Platform: { OS: "web" } }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });

  test("User navigates to More In Social", ({ given, when, then }) => {
    let moreWrapper: ShallowWrapper;
    let instance: More;

    given("I am a User attempting to navigate MoreInSocial Screen", () => {
      moreWrapper = shallow(<More {...screenProps} />);
      expect(moreWrapper).toBeTruthy();
      expect(moreWrapper).toMatchSnapshot();
    });

    when("I navigate to the MoreInSocial Screen", () => {
      instance = moreWrapper.instance() as More;
    });

    then("I can open the modal", async () => {
      let moreInSocialModalButton = moreWrapper.findWhere(
        (node) => node.prop("testID") === "moreInSocialModalButton"
      );
      moreInSocialModalButton.simulate("press");
      expect(moreInSocialModalButton).toBeTruthy();
    });

    then("I can press call menu", async () => {
      let callMenu = moreWrapper.findWhere(
        (node) => node.prop("testID") === "callMenu"
      );
      callMenu.props().onMenuPress();
      expect(callMenu).toBeTruthy();
    });

    then("I can press bot menu", async () => {
      let botMenu = moreWrapper.findWhere(
        (node) => node.prop("testID") === "botMenu"
      );
      botMenu.props().onMenuPress();
      expect(botMenu).toBeTruthy();
    });

    then("I can press file menu", async () => {
      let fileMenu = moreWrapper.findWhere(
        (node) => node.prop("testID") === "fileMenu"
      );
      fileMenu.props().onMenuPress();
      expect(fileMenu).toBeTruthy();
    });

    then("I can press saved menu", async () => {
      let savedMenu = moreWrapper.findWhere(
        (node) => node.prop("testID") === "savedMenu"
      );
      savedMenu.props().onMenuPress();
      expect(savedMenu).toBeTruthy();
    });

    then("I can press alerts menu", async () => {
      let alertsMenu = moreWrapper.findWhere(
        (node) => node.prop("testID") === "alertsMenu"
      );
      alertsMenu.props().onMenuPress();
      expect(alertsMenu).toBeTruthy();
    });

    then("I can press notification menu", async () => {
      let notificationMenu = moreWrapper.findWhere(
        (node) => node.prop("testID") === "notificationMenu"
      );
      notificationMenu.props().onMenuPress();
      expect(notificationMenu).toBeTruthy();
    });

    then("I can press setting menu", async () => {
      let settingslMenu = moreWrapper.findWhere(
        (node) => node.prop("testID") === "settingslMenu"
      );
      settingslMenu.props().onMenuPress();
      expect(settingslMenu).toBeTruthy();
    });
  });

  test("User navigates to MenuComponent", ({ given, when, then }) => {
    let moreWrapper: ShallowWrapper;
    let instance: MenuComponent;

    given("I am a User attempting to navigate MenuComponent", () => {
      moreWrapper = shallow(<MenuComponent {...screenProps} />);
      expect(moreWrapper).toBeTruthy();
      expect(moreWrapper).toMatchSnapshot();
    });

    when("I navigate to the MenuComponent", () => {
      instance = moreWrapper.instance() as MenuComponent;
    });

    then("I can press menu button", async () => {
      let onMenuPreeButton = moreWrapper.findWhere(
        (node) => node.prop("testID") === "onMenuPreeButton"
      );
      onMenuPreeButton.simulate("press");
      expect(onMenuPreeButton).toBeTruthy();
    });
  });
});
