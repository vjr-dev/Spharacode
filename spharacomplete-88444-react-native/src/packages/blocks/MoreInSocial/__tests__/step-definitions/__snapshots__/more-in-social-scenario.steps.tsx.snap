// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreInSocial User navigates to MenuComponent 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": 23,
    }
  }
  testID="onMenuPreeButton"
>
  <Image
    resizeMode="contain"
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "rgb(163,164,165)",
        "fontSize": 13,
        "paddingHorizontal": 23,
        "paddingTop": 5,
      }
    }
  />
</TouchableOpacity>
`;

exports[`MoreInSocial User navigates to More In Social 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../packages/blocks/MoreInSocial/assets/image_background-1.png",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 1334,
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/MoreInSocial/assets/image_background-2.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 1334,
          "width": 750,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "paddingTop": 20,
          }
        }
      >
        More Tab
      </Text>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "marginTop": 20,
          }
        }
        testID="moreInSocialModalButton"
      >
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
          Open Modal
        </Text>
      </TouchableOpacity>
    </ImageBackground>
  </ImageBackground>
</SafeAreaView>
`;
